src/NSwag.AspNetCore/HttpRequestExtension.cs

```diff
@@ -48,29 +51,47 @@ public static string GetBasePath(this IOwinRequest request)
         public static string GetBasePath(this HttpRequest request)
 #endif
         {
-            if (request.Headers.ContainsKey("X-Forwarded-Prefix"))
+            var prefix = TryGetFirstHeader(request.Headers, ForwardedPrefixHeader);
+
+            if (prefix != null)
             {
-                return "/" + request.Headers.TryGetFirstHeader("X-Forwarded-Prefix").Trim('/');
+                return "/" + prefix.Trim('/');
             }
 
-            var basePath = request.Headers.ContainsKey("X-Forwarded-Host") ?
-                new Uri($"http://{request.Headers.TryGetFirstHeader("X-Forwarded-Host")}").AbsolutePath :
-                "";
+            var host = TryGetFirstHeader(request.Headers, ForwardedHostHeader);
+            string basePath;
+
+            if (host != null)
+            {
+                var proto = TryGetFirstHeader(request.Headers, ForwardedProtoHeader) ?? "http";
+                basePath = new Uri($"{proto}://{host}").AbsolutePath.TrimEnd('/');
+            }
+            else
+            {
+                basePath = "";
+            }
 
             if (request.PathBase.HasValue)
             {
-                basePath = basePath.TrimEnd('/') + "/" + request.PathBase.Value;
+                basePath += "/" + request.PathBase.Value;
```

Comment:
Yeah, I moved it to line 67 because on the `else` branch right below that, the `basePath` is assigned an empty string which obviously will never contain a slash.